facebook/OpenBIC
# fby3.5: hd: Read DIMM sensor from BIC

Summary:
- Read DIMM power and temperature from DIMM directly.
- Report DIMM power and temperature to APML.

Dependency: #587 

NOTE!!!
Need to enable "Release SPD Host Control" in BIOS setup menu, it will switch I2C of DIMM from CPU to BIC after post complete.

Test Plan:
- Build code: PASS
- Sensor read: PASS
- Check reported value by AVT: PASS

Log:
Sensor read
root@bmc-oob:~# sensor-util slot1 | grep DIMM
DIMMA Temp                   (0x5) :   40.75 C     | (ok)
DIMMB Temp                   (0x6) :   39.25 C     | (ok)
DIMMC Temp                   (0x7) :   40.50 C     | (ok)
DIMME Temp                   (0x8) :   38.75 C     | (ok)
DIMMG Temp                   (0x9) :   35.50 C     | (ok)
DIMMH Temp                   (0xA) :   33.50 C     | (ok)
DIMMI Temp                   (0xB) :   33.25 C     | (ok)
DIMMK Temp                   (0xC) :   33.50 C     | (ok)
DIMMA Pwr                    (0x33) :    1.00 Watts | (ok)
DIMMB Pwr                    (0x34) :    0.50 Watts | (ok)
DIMMC Pwr                    (0x35) :    1.25 Watts | (ok)
DIMME Pwr                    (0x36) :    1.12 Watts | (ok)
DIMMG Pwr                    (0x37) :    1.62 Watts | (ok)
DIMMH Pwr                    (0x38) :    0.50 Watts | (ok)
DIMMI Pwr                    (0x39) :    0.62 Watts | (ok)
DIMMK Pwr                    (0x3A) :    1.62 Watts | (ok)

Check reported value by AVT tool
[root@localhost AVT_Linux_NDA_2.8.14]# ./AVT -module pmm "get_pmdata()"

PMData:
[Pkg:0], SktPwr(W):53.6225, Determinism slider:Power

[Pkg:0], [INFRASTRUCTURE], TDP:360, PPT_LIMIT:185, TDC_LIMIT_0:175, TDC_LIMIT_1:175, EDC_LIMIT_0:230, EDC_LIMIT_1:230, TJ_MAX:95

[Pkg:0], [FREQUENCIES], FCLK_FREQ:1387.82, UCLK_FREQ:1200, MEMCLK_FREQ:2400

[Pkg:0], [DIMM], Dimm0_Temp:38.25, Dimm0_Power:0.786325, Dimm1_Temp:36.8165, Dimm1_Power:0.637019, Dimm2_Temp:38.5, Dimm2_Power:0.43563, Dimm3_Temp:38, Dimm3_Power:0.316106, Dimm4_Temp:32.5, Dimm4_Power:0.413996, Dimm5_Temp:32.5, Dimm5_Power:1.51442, Dimm6_Temp:34.25, Dimm6_Power:1.19952, Dimm7_Temp:32.75, Dimm7_Power:0.406517
